### ISSUE TITLE
proj5 - clean and tune

### ISSUE THREAD

**Issue by kenny-designs** [kenny-designs/Robot-Projects#69]
As of now, the project is technically done. It could use some cleanup though. main() is looking a bit dirty. Also, world51.cfg trips up the robot some when getting around the obstacle. It does it but it takes a moment. Definitely tune that up so it gets around the object after only 1-3 tries max.

**Comment by kenny-designs** [comment]
Perhaps using the laser or reworking the handleBump() method would be of great help.

**Comment by kenny-designs** [comment]
Made a ton of changes just now. Not gonna merge until tomorrow (more like today considering it is 2am at the time of writing). Everything appears to be working well. One of the biggest changes is a sort of soft implementation of the state design pattern to make the handling of bumper events more flexible. This was perhaps overkill but could prove useful for the final project.

In particular, I added an interface (i.e. a struct with an abstract method denoted by the keyword 'virtual') to Robot.h called BumperEventState. The main feature of this struct is an abstract method called handleBump() that takes a pointer to the robot. So far, there are two structs that implement the BumperEventState interface. They are SimpleBumper and AutoPilot. SimpleBumper has the robot back up, rotate, then move forward in hopes of getting around an unexpected obstacle. AutoPilot on the other hand just invokes the autoPilotLaser() method for a certain number of ticks.

Take a look at follow-plan.cc and the moveToWaypoint() method to see how these new structs are utilized.

Truth be told, I could have just kept extending the original handleBump() method inside of the Robot class. It was getting tedious filling in more and more if statements so I opted for the state design pattern to avoid this (also cause I thought it'd be fun). The original handleBump() method is now completely gone and these three structs have taken its place.
